Python/Product/VSInterpreters/Interpreter/CondaEnvironmentFactoryProvider.cs

```diff
@@ -300,15 +300,17 @@ internal class CondaInfoResult {
             public string RootPrefixFolder = null;
         }
 
-        private void FindCondaEnvironments(List<PythonInterpreterInformation> envs) {
+        private async Task<IReadOnlyList<PythonInterpreterInformation>> FindCondaEnvironments() {
             var mainCondaExePath = _condaLocatorProvider?.FindLocator()?.CondaExecutablePath;
             if (!string.IsNullOrEmpty(mainCondaExePath)) {
-                envs.AddRange(FindCondaEnvironments(mainCondaExePath));
+                return await FindCondaEnvironments(mainCondaExePath);
             }
+            return Enumerable.Empty<PythonInterpreterInformation>().ToList().AsReadOnly();
         }
 
-        private IReadOnlyList<PythonInterpreterInformation> FindCondaEnvironments(string condaPath) {
-            var condaInfoResult = _joinableTaskFactory.Run(() => ExecuteCondaInfoAsync(condaPath));
+        private async Task<IReadOnlyList<PythonInterpreterInformation>> FindCondaEnvironments(string condaPath) {
+            // Make sure to run this on a background thread as it can take a long time.
+            var condaInfoResult = await System.Threading.Tasks.Task.Run(() => ExecuteCondaInfoAsync(condaPath));
```

Comment:
As I recall, it used to be necessary to use `JoinableTaskFactory` in VS (in general, not just here) to avoid deadlocks when two tasks are both scheduled to run on the UI thread, but are awaiting each other. Did that change?

Reply:
It doesn't look like there's two tasks awaiting each other in this case? The main thread is awaiting on the task I kicked off. That thread is not calling back into the main thread AFAICT. 

The reason I moved this to Task.Run instead of JTF.Run. JTF.Run is a synch blocking wait essentially. At least according to the linked bug.

Is there a JTF safe method for kicking off a background thread to do work but not blocking? JTF.RunAsync seems to run the call on the main thread (when I debug it).